### ISSUE TITLE
Audio settings are changed even when not wanted

### ISSUE THREAD

**Issue by DiThi** [OpenVR-Advanced-Settings/OpenVR-AdvancedSettings#85]
When you set in SteamVR not to change the audio devices, sometimes, for some reason I can't reproduce (switching headsets?), it changes the audio to the last used device or whatever.

When SteamVR is set not to change the audio devices, AdvancedSettings shouldn't touch them either.

I'm sure it's AdvancedSettings because I uninstalled it 2 weeks ago and it hasn't happened ever since.

**Comment by ykeara** [comment]
Yes advanced Settings is hitting that API call that is the same one you are using when you use the SteamVR settings.... as such it changes what the "default" is.  If audio devices are added or removed from your system it can also mess up the saved stuff. (advanced settings saved profiles that is). You can use the default profile to always load up a certain config, and profiles to switch between them.

So long story short Its an easy enough fix. but should it be considered the normal behavior? I am open to either interpretation.

**Comment by ykeara** [comment]
I don't remember all behavior around the API and our code. Marking as a bug for now, Need to test API calls and play around with a few options to formalize the behavior, and then discuss what the "correct" behavior of the program should be.

**Comment by DiThi** [comment]
Audio devices are added and removed from my system ALL the time. I use 3 bluetooth headphones and I cycle between them while each battery is depleted. Also devices are added and removed when I disconnect the screen and an external soundcard (to move the laptop around) or when I switch headsets.

Also the Samsung Galaxy audio output only appears when WMR portal is open, so even if I leave that headset plugged all the time it may count as adding/removing devices.

What is the "default profile"? Is it always loaded on startup? Is it no longer loaded if I remove it?

**Comment by DiThi** [comment]
By the way, sorry for the non descriptive title, for some reason GitHub isn't letting me edit it. I wanted to say "Audio settings are changed even when not wanted" or something like that.

**Comment by ykeara** [comment]
The default profile should always apply on start-up, but if the devices change it might prove problematic. (i.e. not apply)... if you delete it, it does remove it.... from the sound of your situation atm, I would change steamVR to "don't change device" and not change it via advanced settings for the time being. It Definitely does need some adjustments, though..

Again without looking at it atm, I believe I find the device index and compare it against the name on apply.... where rather I should do the opposite and that should fix the device not there issue; however, the steamVR behavior is a bit different of a story, so I need to look into that as well.

At any rate I need to finish up the chaperone page changes, and then I want to do Issue #81, then I will try to address this it may or may not make 3.0.0 release, depends on testing and other work.
